adafruit_httpserver.py

```diff
@@ -327,26 +327,37 @@ def start(self, host: str, port: int = 80, root: str = "") -> None:
         )
         self._sock.bind((host, port))
         self._sock.listen(10)
+        self._sock.setblocking(False)  # non-blocking socket
 
     def poll(self):
         """
         Call this method inside your main event loop to get the server to
         check for new incoming client requests. When a request comes in,
         the application callable will be invoked.
         """
-        conn, _ = self._sock.accept()
-        with conn:
-            length, _ = conn.recvfrom_into(self._buffer)
-
-            request = _HTTPRequest(raw_request=self._buffer[:length])
-
-            # If a route exists for this request, call it. Otherwise try to serve a file.
-            route = self.routes.get(request, None)
-            if route:
-                response = route(request)
-            elif request.method == "GET":
-                response = HTTPResponse(filename=request.path, root=self.root_path)
-            else:
-                response = HTTPResponse(status=HTTPStatus.INTERNAL_SERVER_ERROR)
-
-            response.send(conn)
+        try:
+            conn, _ = self._sock.accept()
+            with conn:
+                length, _ = conn.recvfrom_into(self._buffer)
+
+                request = _HTTPRequest(raw_request=self._buffer[:length])
+
+                # If a route exists for this request, call it. Otherwise try to serve a file.
+                route = self.routes.get(request, None)
+                if route:
+                    response = route(request)
+                elif request.method == "GET":
+                    response = HTTPResponse(filename=request.path, root=self.root_path)
+                else:
+                    response = HTTPResponse(status=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+                response.send(conn)
+        except OSError as ex:
+            # handle EAGAIN and ECONNRESET
+            if ex.errno == EAGAIN:
+                # there is no data available right now, try again later".
+                return
+            if ex.errno == ECONNRESET:
+                # connection reset by peer, try again later".
```

Comment:
Did you mean to put "later" in quotes? There's only one quotation mark?

Reply:
Not sure how that " got in there.  Mabe a cut/paste error or something.  I'll fix it.